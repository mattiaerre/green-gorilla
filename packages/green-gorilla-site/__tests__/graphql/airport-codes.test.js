const airportCodes = require('../../src/graphql/airport-codes');

describe('airport-codes', () => {
  test('API', () => {
    expect(airportCodes).toMatchSnapshot();
  });

  const TORINO = 'Torino';
  const MALTA = 'Malta';
  const GATWICK_ID = '502';

  const scenarios = [
    {
      description: `filter by city: ${TORINO}`,
      query: { city: TORINO }
    },
    {
      description: `filter by city: ${TORINO}`,
      query: { city: TORINO, country: '' }
    },
    {
      description: `filter by country: ${MALTA}`,
      query: { country: MALTA }
    },
    {
      description: `filter by country: ${MALTA}`,
      query: { city: '', country: MALTA }
    },
    {
      description: `filter by id: ${GATWICK_ID}`,
      query: { id: GATWICK_ID }
    }
  ];

  scenarios.forEach(({ description, query }) => {
    test(description, () => {
      expect(airportCodes.resolver(query)).toMatchSnapshot();
    });
  });
});

// url: http://localhost:9000/graphql?query=%7B%0A%20%20airportCodes(city%3A%22Milan%22)%20%7B%0A%20%20%20%20id%0A%20%20%20%20name%0A%20%20%20%20city%0A%20%20%20%20country%0A%20%20%20%20iata%0A%20%20%20%20icao%0A%20%20%7D%0A%7D%0A
