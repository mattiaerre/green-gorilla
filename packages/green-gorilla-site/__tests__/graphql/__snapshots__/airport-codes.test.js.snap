// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`airport-codes API 1`] = `
Object {
  "TYPE_NAME": "AirportCode",
  "resolver": [Function],
  "schema": "
  type AirportCode {
    id: String,
    name: String,
    city: String,
    country: String,
    iata: String,
    icao: String
  }
",
}
`;

exports[`airport-codes filter by city: Torino 1`] = `
Array [
  Object {
    "altitude": "989",
    "city": "Torino",
    "country": "Italy",
    "dst": "E",
    "iata": "TRN",
    "icao": "LIMF",
    "id": "1526",
    "latitude": "45.200761",
    "longitude": "7.649631",
    "name": "Torino",
    "timezone": "1",
    "tz": "Europe/Rome",
  },
]
`;

exports[`airport-codes filter by city: Torino 2`] = `
Array [
  Object {
    "altitude": "989",
    "city": "Torino",
    "country": "Italy",
    "dst": "E",
    "iata": "TRN",
    "icao": "LIMF",
    "id": "1526",
    "latitude": "45.200761",
    "longitude": "7.649631",
    "name": "Torino",
    "timezone": "1",
    "tz": "Europe/Rome",
  },
]
`;

exports[`airport-codes filter by country: Malta 1`] = `
Array [
  Object {
    "altitude": "300",
    "city": "Malta",
    "country": "Malta",
    "dst": "E",
    "iata": "MLA",
    "icao": "LMML",
    "id": "1606",
    "latitude": "35.857497",
    "longitude": "14.4775",
    "name": "Luqa",
    "timezone": "1",
    "tz": "Europe/Malta",
  },
  Object {
    "altitude": "300",
    "city": "Gozo",
    "country": "Malta",
    "dst": "E",
    "iata": "GZM",
    "icao": "LMMG",
    "id": "4005",
    "latitude": "36.027222",
    "longitude": "14.272778",
    "name": "Xewkija Heliport",
    "timezone": "1",
    "tz": "Europe/Malta",
  },
]
`;

exports[`airport-codes filter by country: Malta 2`] = `
Array [
  Object {
    "altitude": "300",
    "city": "Malta",
    "country": "Malta",
    "dst": "E",
    "iata": "MLA",
    "icao": "LMML",
    "id": "1606",
    "latitude": "35.857497",
    "longitude": "14.4775",
    "name": "Luqa",
    "timezone": "1",
    "tz": "Europe/Malta",
  },
  Object {
    "altitude": "300",
    "city": "Gozo",
    "country": "Malta",
    "dst": "E",
    "iata": "GZM",
    "icao": "LMMG",
    "id": "4005",
    "latitude": "36.027222",
    "longitude": "14.272778",
    "name": "Xewkija Heliport",
    "timezone": "1",
    "tz": "Europe/Malta",
  },
]
`;

exports[`airport-codes filter by id: 502 1`] = `
Array [
  Object {
    "altitude": "202",
    "city": "London",
    "country": "United Kingdom",
    "dst": "E",
    "iata": "LGW",
    "icao": "EGKK",
    "id": "502",
    "latitude": "51.148056",
    "longitude": "-0.190278",
    "name": "Gatwick",
    "timezone": "0",
    "tz": "Europe/London",
  },
]
`;
